### PR TITLE
Spectral locator

### PR DESCRIPTION
**Pull request by Ilya246**
## About the PR
see https://github.com/space-wizards/space-station-14/pull/32323 for all information, this is a direct port of that PR; PRing it here as well since wizden reviews usually take a long time and it has a fair chance to get denied there

**Changelog**

:cl:
- add: Spectral locator, rare maintenance loot that lets you know if any ghosts are nearby. Also found in the chaplain's vendor.

## Summary by CodeRabbit

- **New Features**
	- Introduced the `SpectralComponent` to identify ghostly entities.
	- Added a new item, `SpectralLocator`, enhancing gameplay with spectral locator mechanics.
	- Added support for spectral representation in player mobs.
	- Introduced three new entities related to the spectral locator system with varying power states.

- **Documentation**
	- Added metadata for the spectral locator resource, detailing its properties and states.

### CODE REVIEW

File: Content.Shared/Goobstation/Ghost/SpectralComponent.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Ghost;
```
Comment (Piras314) [inline Content.Shared/Goobstation/Ghost/SpectralComponent.cs:3]: namespace Content.Shared.Goobstation.Ghost;

File: Resources/Prototypes/Goobstation/Entities/Objects/Fun/spectral_locator.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+- type: entity
+  id: SpectralLocatorUnpowered
+  parent: BaseItem
+  name: spectral locator
+  description: Appears to be a modified anomaly locator. Seems very old.
+  suffix: Unpowered
+  components:
+  - type: Sprite
+    sprite: Objects/Fun/spectrallocator.rsi
```
Comment (Piras314) [inline Resources/Prototypes/Goobstation/Entities/Objects/Fun/spectral_locator.yml:9]: Goobstation/Objects/Fun/spectrallocator.rsi (you already put it there but forgot to update this)

**Review by Piras314 — CHANGES_REQUESTED** [review]
see previous review comments + I think it's a good idea to add it to the chaplain loadout

**Comment by Ilya246** [comment]
> see previous review comments + I think it's a good idea to add it to the chaplain loadout

i'll put one into chaplain's locker

**Comment by Ilya246** [comment]
compiled and tested ingame just to be sure, works

**Review by Piras314 — APPROVED** [review]

**Review by whateverusername0 — APPROVED** [review]

**Pull request merged**